baseplate/lib/metrics.py

```diff
@@ -313,21 +313,19 @@ def send(self, elapsed: float, sample_rate: float = 1.0) -> None:
         want to report the result.
         :param elapsed: The elapsed time in seconds to report.
         """
-
+        serialized = self.name
         formatted_tags = _format_tags(self.tags)
         if formatted_tags:
-            parts = [self.name + formatted_tags.encode() + f":{(elapsed * 1000.0):g}|ms".encode()]
-        else:
-            parts = [self.name + f":{(elapsed * 1000.0):g}|ms".encode()]
+            serialized += formatted_tags
+        serialized += f":{(elapsed * 1000.0):g}|ms".encode()
         if sample_rate < 1.0:
             sampling_info = f"@{sample_rate:g}".encode()
-            parts.append(sampling_info)
-        serialized = b"|".join(parts)
+            serialized = b"|".join([serialized, sampling_info])
         self.transport.send(serialized)
 
     def update_tags(self, tags: Dict) -> None:
         assert not self.stopped
-        self.tags.update(tags)
+        self.tags = tags
```

Comment:
I think I wrote this under the assumption that the tags were being kept in parallel on the span, so any updates could just completely overwrite the existing tags. However, this may not work for users and other use cases so I'll change it back